docs/development/adding-a-package-manager.md

```diff
@@ -14,11 +14,23 @@ Common application logic for Renovate, not specific to particular managers, usua
 
 The manager's `index.ts` file supports the following values/functions:
 
-- extractPackageFile
-- extractAllPackageFiles
-- getRangeStrategy (optional)
-- language (optional)
-- supportsLockFileMaintenance (optional)
+| Value/function                | Optional |
+| ----------------------------- | -------- |
+| `bumpPackageVersion`          | yes      |
+| `extractPackageFile`          |          |
+| `extractAllPackageFiles`      |          |
+| `getRangeStrategy`            | yes      |
+| `language`                    | yes      |
+| `supportsLockFileMaintenance` | yes      |
+| `updateArtifacts`             | yes      |
+| `updateDependency`            | yes      |
+| `updateLockedDependency`      | yes      |
```

Comment:
How do we want to order the table and the headings:

- Strict `A-Z`: messes up `optional` column
- `aync` then `normal` functions: messes op the `optional` column
- `optional:yes` first, then by `A-Z`: causes a break in the `A-Z` ordering, but keeps optional things together
- Keep things as they are now

Reply:
Let's keep strict A-Z. As we leave optional empty if it is false, required functions are easy identifiable.

Reply:
I'll let you move the heading and the table items and create the `async` column. 😉

Reply:
I have decided to keep the order as only `extractPackageFile` and `extractAllPackageFiles` would be switched and we prefer the first one.

Reply:
That's okay!